minetest/minetest
# Night vision

This is early work to enable ambient light in caves and features like night vision in the games. Fixes #6509 and fixes #8890.

This is shader-specific implementation, but it should be possible to add similar code for fixed rendering pipeline.

## Plan
  - [x] Collect feedback
  - [x] Add server-side API to control the ambient brightness
  - [x] Add ambient color, see #8890
  - [ ] Integrate with set_lighting
  - [ ] Add support for particles

This PR is Work in Progress

## How to test

* Install WorldEdit mod
* Start any game in single player get all privileges and go underground (fly + noclip).
* Type `//lua minetest.get_player_by_name('singleplayer'):set_ambient_light({brightness=0.5, color_tint="#ffd480"})`.
* You should now see all the caves with an orange tint.
